flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgram.java

```diff
@@ -440,7 +440,7 @@ private static String getEntryPointClassNameFromJar(URL jarFile)
     }
 
     @Nullable
-    private static URL loadJarFile(File jar) throws ProgramInvocationException {
+    private static URL checkJarFile(File jar) throws ProgramInvocationException {
```

Comment:
I see on line 453 an existing call to `checkJarFile(jarFileUrl); ` So the method is now recursive, when before the fix it was not . Am I missing something?

Reply:
Yes, I know it. I think the overriding method is not a problem.